snapcraft/internal/elf.py

```diff
@@ -38,7 +38,7 @@
 class ElfFile:
     """ElfFile represents and elf file on a path and its attributes."""
 
-    def __init__(self, *, path: str, is_executable: bool) -> None:
+    def __init__(self, *, path: str, is_executable: bool=True) -> None:
```

Comment:
This is probably a nit that you can ignore, but I'm not very happy with this `is_executable` attribute. Wouldn't it be better to remove the attribute, and add an `is_executable()` method that checks the headers?
The downside is that you would have to use magic here again. Or maybe, we move all the checks inside this constructor. If it's not an elf, init raises an exception. And we could have also a method for `is_dynamically_linked()`.

Reply:
I'll let @kyrofa weigh in. I am not a fan of either.

Reply:
as in, either your or my proposal :wink:

Reply:
Yeah I agree with @elopio, it'd be nice to see this encapsulated a little more, although I'd like to see it either lazy or in `__init__()`. I suspect you wouldn't want to load up magic every time though... I suppose you could pass it in.

Reply:
heh, this was previous PR material ;)